sklearn/manifold/tests/test_mds.py

```diff
@@ -55,10 +55,103 @@ def test_smacof_error():
         mds.smacof(sim, init=Z, n_init=1)
 
 
+def test_svd():
+    # Test svd using example data from "An Introduction to MDS"
+    # Florian Wickelmaier, p 11
+    sim = np.array([[0, 93, 82, 133],
+                    [93, 0, 52, 60],
+                    [82, 52, 0, 111],
+                    [133, 60, 111, 0]])
+
+    X, stress = mds.svd_scaler(sim, n_components=2)
+    X_true_1 = np.array([[-62.831, -32.97448],
+                         [18.403, 12.02697],
+                         [-24.960, 39.71091],
+                         [69.388, -18.76340]])
+    X_true_2 = np.copy(X_true_1)
+    X_true_2[:, 0] *= -1
+
+    # Signs of columns are dependent on signs of computed eigenvectors
+    # which are arbitrary and meaningless
+    assert(np.allclose(X, X_true_1)
+           or np.allclose(X, -X_true_1)
+           or np.allclose(X, X_true_2)
+           or np.allclose(X, -X_true_2))
+
+
+def test_svd_error():
+    # Non symmetric (dis)similarity matrix:
+    sim = np.array([[0, 5, 9, 4],
+                    [5, 0, 2, 2],
+                    [3, 2, 0, 1],
+                    [4, 2, 1, 0]])
+
+    with pytest.raises(ValueError):
+        mds.svd_scaler(sim)
+
+    # Non squared (dis)similarity matrix:
+    sim = np.array([[0, 5, 9, 4],
+                    [5, 0, 2, 2],
+                    [4, 2, 1, 0]])
+
+    with pytest.raises(ValueError):
+        mds.svd_scaler(sim)
+
+    # Non Euclidean (dis)similarity matrix:
+    sim = np.array([[0, 12, 3, 4],
+                    [12, 0, 2, 2],
+                    [3, 2, 0, 1],
+                    [4, 2, 1, 0]])
+
+    with pytest.raises(ValueError,
+                       match="Dissimilarity matrix must be euclidean"):
+        mds.svd_scaler(sim)
+
+
+def test_MDS_error():
```

Comment:
Let's catch the expected error message with `match=...`

Reply:
Done